python/lsst/rubintv/production/plotting/latissNightReportPlots.py

```diff
@@ -0,0 +1,1000 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.dates as md
+
+from lsst.summit.utils.utils import getAirmassSeeingCorrection, getFilterSeeingCorrection
+
+from .nightReportPlotBase import LatissPlot
+
+# any classes added to __all__ will automatically be added to the night
+# report channel, with each being replotted for each image taken.
+__all__ = ['ZeroPointPlot',
+           'SkyMeanPlot',
+           'PsfFwhmPlot',
+           'SourceCountsPlot',
+           'PsfE1Plot',
+           'PsfE2Plot',
+           'TelescopeAzElPlot',
+           'MountMotionPlot',
+           'MountMotionVsZenith',
+           'SkyMeanVsSkyRms',
+           'PsfVsSkyRms',
+           'PsfVsZenith',
+           'PsfVsMountMotion',
+           ]
+
+# TODO: get these colours from somewhere else
+gcolor = 'mediumseagreen'
+rcolor = 'lightcoral'
+icolor = 'mediumpurple'
+
+
+class ZeroPointPlot(LatissPlot):
+    _PlotName = 'per-band-zeropoints'
+    _PlotGroup = 'Photometry'
+
+    def __init__(self,
+                 dayObs,
+                 locationConfig=None,
+                 uploader=None):
+        super().__init__(dayObs=dayObs,
+                         plotName=self._PlotName,
+                         plotGroup=self._PlotGroup,
+                         locationConfig=locationConfig,
+                         uploader=uploader)
+
+    def plot(self, nightReport, metadata, ccdVisitTable):
+        """Create the zeropoint plot.
+
+        Parameters
+        ----------
+        nightReport : `lsst.rubintv.production.nightReport.NightReport`
+            The night report for the current night.
+        metadata : `pandas.DataFrame`
+            The front page metadata, as a dataframe.
+        ccdVisitTable : `pandas.DataFrame`
+            The visit summary table for the current day.
+
+        Returns
+        -------
+        success : `bool`
+            Did the plotting succeed, and thus upload should be performed?
+        """
+        for item in ['Zeropoint', 'Filter']:
+            if item not in metadata.columns:
+                msg = f'Cannot create {self._PlotName} plot as required item {item} is not in the table.'
+                self.log.warning(msg)
+                return False
+
+        # TODO: get a figure you can reuse to avoid matplotlib memory leak
```

Comment:
You've got a lot of these comments, so I'll only comment on it once here:

- Instead of going all the way to a reusable `Figure`, it'd be worth trying to: get a `Figure` via `matplotlib.figure.Figure`, get an `Axes` (or a few) from `Figure.subplots`, and then call methods on `Axes` to do the plotting.  That requires far less global state (maybe none?) so it should be better at cleaning up after itself, and the `Axes` methods are almost all drop-in replacements for the `matplotlib.pyplot` functions (in m experience, you just have to put `set_` in front of `xlim`, `ylim`, `title`, and things like that).

- If that doesn't help, try `gc.collect().`